client/src/dataset/components/ScanSize.tsx

```diff
@@ -0,0 +1,70 @@
+import { Field } from "formik";
+import * as React from "react";
+import { Button, Form, FormFieldProps } from "semantic-ui-react";
+
+interface ScanSizeProps {
+    value: string,
+    minScan: number,
+    maxScan: number,
+    setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+}
+
+const ScanSize: React.FC<ScanSizeProps> = ({ value, minScan, maxScan, setFieldValue }) => {
+
+    const [scanSize, setScanSize] = React.useState(value.split(",").length>=minScan? value.split(","): new Array(minScan).fill(""));
+
+    const scanRefsArray = React.useRef<FormFieldProps[]>([]);
+
+    const onScanSizeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const idx = parseInt(e.target.name.split("_")[2], 10);
+      scanSize[idx] = e.target.value.toString();
+      setScanSize(scanSize);
+      setFieldValue("scan_size", scanSize.toString());
+    };
+
+    const onCommaPress = (e: KeyboardEvent) => {
+        const idx = parseInt((e.target as HTMLInputElement).name.split("_")[2], 10);
+        if(e.keyCode === 188){
+            idx===(scanSize.length-1)? newScanDim() : scanRefsArray.current[idx+1].focus();
+        }
+    }
+
+    const newScanDim = () => {
+      if(scanSize.length < maxScan) {
+        scanSize.push("");
+        setScanSize(scanSize);
+        setFieldValue("scan_size", scanSize.toString());
+      }
+    }
+
+    React.useEffect(() => {
+      if(scanSize.length > minScan) {
+        scanRefsArray.current[scanSize.length-1].focus();
+      }
+    }, [scanSize.length, minScan]);
+
+    const delScanDim = () => {
+      if(scanSize.length > minScan) {
+        scanSize.pop();
+        setScanSize(scanSize);
+        setFieldValue("scan_size", scanSize.toString());
+      }
+    }
+
+    return (
+      <>
+        <Form.Group>
+          {scanSize.map((val, idx) => {
+            const scanRef = (ref:FormFieldProps) => { scanRefsArray.current[idx] = ref; }
+            return <Form.Field key={idx} width={2}><Field name={"scan_size_" + idx} id={"id_scan_size_" + idx} type="number" value={val} innerRef={scanRef} onChange={onScanSizeChange} onKeyDown={onCommaPress} /></Form.Field>
```

Comment:
Hmm, passing information via the `name` like this is a bit brittle - if for some reason the `name` needs to change, this logic will break. For example, let's say you want to use `ScanSize` also for a field called `some_other_tuple` - now the index won't be at position 2 anymore.

I would suggest to extract the "part" as a separate component, so for each tuple element, you would use this component, like this:
```
<ScanSizePart key={idx} value={val} onChange={...} {...etc} />
```

Then, inside that component, you know the `idx` and can pass it up to the change handler in `ScanSize`, something like this:

```
const ScanSizePart = ({ idx, onChange, value }) => {
   const onPartChange = (e) => {
      onChange(idx, e.target.value);
   }

   return <input type='number' onChange={onPartChange} value={value} />; 
}
```

Then, `ScanSize` can still do its state management in `onScanSizeChange`, just without needing to parse the `name` anymore.

I'm also not sure it is the right thing to use the Formik `<Field />` for the parts - as the documentation states, "<Field /> will automagically hook up inputs to Formik. It uses the name attribute to match up with Formik state." - so you end up with values for the parts in the formik state. Maybe just a plain input would work here? The hooking up to formik state part you do manually anyways, using `setFieldValue`...